eureka-client/src/main/java/com/netflix/appinfo/ApplicationInfoManager.java

```diff
@@ -219,6 +219,21 @@ public void refreshDataCenterInfoIfRequired() {
             builder.setHostName(newAddress).setIPAddr(newIp).setDataCenterInfo(config.getDataCenterInfo());
             instanceInfo.setIsDirty();
         }
+
+        String existingSpotInstanceAction  = instanceInfo.getMetadata().get(AmazonInfo.MetaDataKey.spotInstanceAction.name);
```

Comment:
(mostly due to our bad naming) I think this is the wrong metadata map to get here, you want the metadata from the instanceInfo.getDataCenterInfo(), if it is an instance of AmazonInfo.

Reply:
Right. Fixed.